transform-message-formats/tests/test.bal

```diff
@@ -0,0 +1,155 @@
+import ballerina/http;
+import ballerina/test;
+
+final http:Client cl = check new (string `http://localhost:${port}/healthcare`);
+
+const GRAND_OAKS_HOSPITAL = "grand oak community hospital";
+const CLEMENCY_MEDICAL_CENTER = "clemency medical center";
+const PINE_VALLEY_COMMUNITY_HOSPITAL = "pine valley community hospital";
+const DEFAULT_DOCTOR = "thomas collins";
+
+@test:Config
+function testSuccessfullReservation() returns error? {
+    ReservationResponse resp = check cl->/categories/surgery/reserve.post({
+        name: "John Doe",
+        dob: "1940-03-19",
+        ssn: "234-23-525",
+        address: "California",
+        phone: "8770586755",
+        email: "johndoe@gmail.com",
+        doctor: "thomas collins",
+        hospital_id: "grandoaks",
+        hospital: "grand oak community hospital",
+        card_no: "7844481124110331",
+        appointment_date: "2025-04-02"
+    });
+    ReservationResponse expResp = getSuccessAppointmentResponse(GRAND_OAKS_HOSPITAL);
+    test:assertEquals(resp, expResp, "Response mismatched");
+}
+
+@test:Config
+function testInvalidDoctor() {
+    ReservationResponse|http:ClientError resp = cl->/categories/surgery/reserve.post({
+        name: "John Doe",
+        dob: "1940-03-19",
+        ssn: "234-23-525",
+        address: "California",
+        phone: "8770586755",
+        email: "johndoe@gmail.com",
+        doctor: "thomas chandler",
+        hospital_id: "grandoaks",
+        hospital: "grand oak community hospital",
+        card_no: "7844481124110331",
+        appointment_date: "2025-04-02"
+    });
+
+    if resp !is http:ClientRequestError {
+        test:assertFail("expected an http:ClientRequestError, found " + (typeof resp).toString());
+    }
+
+    test:assertEquals(resp.message(), "Not Found");
+    var detail = resp.detail();
+    test:assertEquals(detail.statusCode, http:STATUS_NOT_FOUND);
+    test:assertEquals(detail.body, "Reservation failed. Wrong hospital or doctor");
+}
+
+@test:Config
+function testBadRequest() {
```

Comment:
Can we improve the name? Other tests explain what exactly is being tested vs this is the response?